src/backend/api/Fusion.Resources.Domain/Commands/Personnel/CreateOrUpdatePersonNote.cs

```diff
@@ -0,0 +1,104 @@
+﻿using FluentValidation;
+using Fusion.Resources.Database;
+using Fusion.Resources.Database.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+#nullable enable 
+
+namespace Fusion.Resources.Domain.Commands
+{
+    public class CreateOrUpdatePersonNote : TrackableRequest<QueryPersonNote> 
+    {
+        private CreateOrUpdatePersonNote(Guid? id, string? content, bool isShared)
+        {
+            NoteId = id;
+            Content = content;
+            IsShared = isShared;
+        }
+
+        public Guid? NoteId { get; }
+        public string? Content { get; }
+        public bool IsShared { get; }
+
+        public string? Title { get; set; }
+        public Guid AzureUniqueId { get; set; }
+
+        public CreateOrUpdatePersonNote OnUser(Guid azureUniqueId)
+        {
+            AzureUniqueId = azureUniqueId;
+            return this;
+        }
+        public CreateOrUpdatePersonNote WithTitle(string? title)
+        {
+            Title = title;
+            return this;
+        }
+
+        public static CreateOrUpdatePersonNote CreateNew(string? content, bool isShared) => new CreateOrUpdatePersonNote(null, content, isShared);
```

Comment:
I added a validator that ensures that the azure id is not empty.. 
Tried to reduce the method signature as it would rather long.. Could maybe switch it around to have the shared as a method..